static-site/pages/blog.jsx

```diff
@@ -0,0 +1,32 @@
+import { useRouter } from "next/router";
+import { useEffect} from 'react';
+import { getBlogPosts } from "../src/util/blogPosts";
+
+
+/**
+ * Default export is a component which immediately redirects to the latest blog
+ * post. Note that we could achieve the same end by adding redirects to
+ * `next.config.mjs` but we run into issues (performance & technical) when
+ * running getBlogPosts there.
+ */
+export default function Index({redirectTo}) {
+  const router = useRouter();
+  useEffect(() => {
+    if (redirectTo) router.push(redirectTo);
+  }, [redirectTo, router])
+  return null;
```

Comment:
Good find! Fixed (in forthcoming force push) by using `router.replace`. (The buggy behaviour was because we would redirect by _adding_ a new page onto the navigation stack so when you go "back" you then hit the same code which adds the page you were just on.)

This also occurred in the groups app (`/` to `/groups`) which is harder to test but has also been fixed.